File: kotlin/processor/src/main/kotlin/io/todak/conveniencepromotion/processor/domain/EventGoodsRepository.java
package io.todak.conveniencepromotion.processor.domain;

import io.todak.conveniencepromotion.domain.eventgoods.EventGoodsEntity;
import io.todak.conveniencepromotion.domain.eventgoods.StoreType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface EventGoodsRepository extends JpaRepository<EventGoodsEntity, Long> {

    Optional<EventGoodsEntity> findByProductNameAndEventMonthAndStore(String productName, String eventMonth, StoreType store);

}

File: common/src/main/kotlin/io/todak/conveniencepromotion/eventgoods/EventGoodsRepository.kt
package io.todak.conveniencepromotion.eventgoods

import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface EventGoodsRepository : JpaRepository<EventGoodsEntity, Long> {
    fun findByProductNameAndEventMonthAndStore(
        productName: String,
        eventMonth: String,
        store: String
    ): Optional<EventGoodsEntity>

}